core-ui/src/shared/hooks/useMounted.js

```diff
@@ -0,0 +1,13 @@
+import { useEffect, useRef } from 'react';
+
+// prevents `setState` on onmounted component
+export function useMounted() {
+  const mounted = useRef(false);
+
+  useEffect(() => {
+    mounted.current = true;
+    return () => (mounted.current = false);
+  }, []);
+
+  return mounted;
```

Comment:
There;s a hook for that already, the filename is 'useComponentDidMount'

Reply:
Actually it's not the same? It doesn't return "mountedness" status (which is required here), it just runs a function once.

Reply:
It's a fairly similar functionality. Why is it better to have it separated?